PyInstaller/utils/hooks/__init__.py

```diff
@@ -173,48 +173,54 @@ def get_pyextension_imports(modname):
     return module_imports
 
 
-def qt4_plugins_dir(ns='PyQt4'):
-    qt4_plugin_dirs = eval_statement(
-        "from %s.QtCore import QCoreApplication;"
-        "app=QCoreApplication([]);"
-        # For Python 2 print would give "<PyQt4.QtCore.QStringList
-        # object at 0x....>", so we need to convert each element separately
-        "str=getattr(__builtins__, 'unicode', str);" # for Python 2
-        "print([str(p) for p in app.libraryPaths()])" % ns)
-    if not qt4_plugin_dirs:
-        logger.error('Cannot find %s plugin directories' % ns)
-        return ""
-    for d in qt4_plugin_dirs:
-        if os.path.isdir(d):
-            return str(d)  # must be 8-bit chars for one-file builds
-    logger.error('Cannot find existing %s plugin directory' % ns)
-    return ""
-
-
-def qt4_phonon_plugins_dir(ns='PyQt4'):
-    qt4_plugin_dirs = eval_statement(
-        "from PyQt4.QtGui import QApplication;"
-        "app=QApplication([]); app.setApplicationName('pyinstaller');"
-        "from PyQt4.phonon import Phonon;"
-        "v=Phonon.VideoPlayer(Phonon.VideoCategory);"
-        # For Python 2 print would give "<PyQt4.QtCore.QStringList
+def qt_plugins_dir(namespace):
+    """
+    Return list of paths searched for plugins
+
+    :param namespace: Import namespace, i.e., PyQt4, PyQt5, or PySide
+
+    :return: Plugin directory paths
+    """
+    if namespace not in ['PyQt4', 'PyQt5', 'PySide']:
+        raise Exception('Invalid namespace: {0}'.format(namespace))
+    qt_plugin_paths = []
+    paths = eval_statement("""
+        from {0}.QtCore import QCoreApplication;
+        app = QCoreApplication([]);
+        # For Python 2 print would give <PyQt4.QtCore.QStringList
```

Comment:
Did you test this code? This should fail with IndentationError: in the second line.